src/modules/iotjs_module_mqtt.c

```diff
@@ -565,8 +587,175 @@ JS_FUNCTION(MqttHandle) {
     case SUBSCRIBE:
     case UNSUBSCRIBE:
     case PINGREQ:
+      return MQTT_ERR_UNALLOWED_PACKET;
+  }
+
+  return 0;
+}
+
+
+static void iotjs_mqtt_concat_buffers(iotjs_mqttclient_t *mqttclient,
+                                      iotjs_bufferwrap_t *buff_recv) {
+  char *tmp_buf = mqttclient->local_buffer;
+  mqttclient->local_buffer = (char *)malloc(
+      sizeof(char) * (buff_recv->length + mqttclient->buffer_length));
```

Comment:
the order is `mqttclient->buffer_length + buff_recv->length`. I would assign it to a local variable and just write 
`mqttclient->buffer_length = local_variable` below. Assign is faster tham +=